pkg/server/datastore/sqlstore/sqlstore_test.go

```diff
@@ -641,148 +642,116 @@ func (s *PluginSuite) TestBundlePrune() {
+	})
 }
 
 func (s *PluginSuite) TestRevokeX509CA() {
 	t := s.T()
 
-	// Setup
-	unusedKey := testkey.NewRSA2048(t)
-
-	caCertPublicKeyRaw, err := x509.MarshalPKIXPublicKey(s.cacert.PublicKey)
-	require.NoError(t, err)
+	// SubjectKeyID
+	certID := x509util.SubjectKeyIDToString(s.cert.SubjectKeyId)
 
-	// Tainted public key on raw format
-	certPublicKeyRaw, err := x509.MarshalPKIXPublicKey(s.cert.PublicKey)
-	require.NoError(t, err)
+	// Bundle not found
+	t.Run("bundle not found", func(t *testing.T) {
+		err := s.ds.RevokeX509CA(ctx, "spiffe://foo", "foo")
+		spiretest.RequireGRPCStatus(t, err, codes.NotFound, _notFoundErrMsg)
+	})
 
+	// Create new bundle with two cert (one valid and one expired)
 	keyForMalformedCert := testkey.NewEC256(t)
 	malformedX509 := &x509.Certificate{
 		PublicKey: keyForMalformedCert.PublicKey,
 		Raw:       []byte("no a certificate"),
 	}
-
-	// Create new bundle with two cert (one valid and one expired)
 	bundle := bundleutil.BundleProtoFromRootCAs("spiffe://foo", []*x509.Certificate{s.cert, s.cacert, malformedX509})
-	bundle.X509TaintedKeys = []*common.X509TaintedKey{
-		{PublicKey: []byte("foh")},
-	}
-
-	// Bundle not found
-	err = s.ds.RevokeX509CA(ctx, "spiffe://foo", unusedKey.Public())
-	spiretest.RequireGRPCStatus(t, err, codes.NotFound, _notFoundErrMsg)
-
-	_, err = s.ds.CreateBundle(ctx, bundle)
+	_, err := s.ds.CreateBundle(ctx, bundle)
 	require.NoError(t, err)
 
-	// Bundle contains a malformed tainted key
-	err = s.ds.RevokeX509CA(ctx, "spiffe://foo", unusedKey.PublicKey)
-	spiretest.RequireGRPCStatusHasPrefix(t, err, codes.Internal, "failed to parse tainted Key:")
+	//
+	t.Run("Bundle contains a malformed certificate", func(t *testing.T) {
+		err := s.ds.RevokeX509CA(ctx, "spiffe://foo", "foo")
+		spiretest.RequireGRPCStatusHasPrefix(t, err, codes.Internal, "failed to parse root CA: x509: malformed certificate")
+	})
 
-	// Remove malformed tainted key
-	bundle.X509TaintedKeys = []*common.X509TaintedKey{}
+	// Remove malformed certificate
+	bundle = bundleutil.BundleProtoFromRootCAs("spiffe://foo", []*x509.Certificate{s.cert, s.cacert})
 	_, err = s.ds.UpdateBundle(ctx, bundle, nil)
 	require.NoError(t, err)
 
-	// // No root CA is using provided key
-	// err = s.ds.RevokeX509CA(ctx, "spiffe://foo", unusedKey.PublicKey)
-	// spiretest.RequireGRPCStatus(t, err, codes.NotFound, "no root CA found with provided public key")
+	t.Run("No root CA is using provided skID", func(t *testing.T) {
+		err := s.ds.RevokeX509CA(ctx, "spiffe://foo", "foo")
+		spiretest.RequireGRPCStatus(t, err, codes.NotFound, "no root CA found with provided subject key ID")
+	})
 
-	// No able to revoke untainted bundles
-	err = s.ds.RevokeX509CA(ctx, "spiffe://foo", s.cert.PublicKey)
-	spiretest.RequireGRPCStatus(t, err, codes.InvalidArgument, "it is not possible to revoke an untainted root CA")
+	t.Run("Unable to revoke untainted bundles", func(t *testing.T) {
```

Comment:
It would be good to check that the sequence number of the bundle remains the same for those cases that fail.